### DISCUSSION TITLE
Pager Duty Message Body Templating

### DISCUSSION THREAD

**Discussion by filipdulic** [grafana/grafana#59329] [Alerting]
Current PagerDuty Contact Point `Optional PagerDuty settings` only offer the use of templates in the `Summery` field which only templates the Title of the PagerDuty alert, while the body of the message is the default implementation, and cannot be changed through the UI ( or it's at least not obvious how to ).

This is a request to add one more optional setting with for message text that would use templates, similar to how Slack Contact Point has `Title` and `Text Body` with templates.

**Comment by filipdulic** [comment]
Bump

**Reply by jquick** [reply]
you can do this now, its kinda hard to find, but its in the `details` section of the contact point. You can override the default like:

`firing` -> `{{ template "pd" . }}`

**Reply by jquick** [reply]
Here are the default options you can override:
https://github.com/grafana/alerting/blob/main/receivers/pagerduty/config.go#L20

**Reply by alexralph** [reply]
I am trying to follow how this is done and cannot make it work. My attempts have resulted in a 400 Bad Request returned from PagerDuty.
I need to map certain label kv pairs from the firing object into the `custom_details` object in the Alert object that is sent via the PagerDuty Contact Point to PagerDuty, e.g.:
```
{
  "payload": {
    "custom_details": {  
      "firing": { alert kv's here }
    }
  }
}
```
I want to re-map this into:
```
{
  "payload": {
    "custom_details": { alert kv's here }
} 
```

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
